### PR TITLE
Setup database service for dev and prod, closes #18

### LINKED ISSUE #18: Set up project database service
We need a service on the backend which reads protected environment variables (set in the deployment platform, or in `.env.development` files which are NOT checked into git), and authenticated with cloud atlas to read and write data. This is a rather abstract idea that we'll want to vary based on the environment we're in.

## Acceptance criteria.

- We want a way to differentiate between production and development databases. This can be done in several ways, but most commonly by looking at `process.env.NODE_ENV`
- We only want to write to development databases  if `NODE_ENV !== 'production'`, therefore we might prefix all these DBs with `dev_`
- A common way of handling stuff like this can be done with the config package: https://www.npmjs.com/package/config. This leads to multiple configuration files such as development.js, staging.js, production.js which hold different config values for those environments.

### PR DESCRIPTION
**Pull request by christopher-dodge**
- Add connection to cloud Mongo (Atlas) and local mongo
- Create dev_storie database for development and storie database for production

### CODE REVIEW

File: server/database/connection.js [hunk 1]
```diff
@@ -1,35 +1,44 @@
...
-const URI               = process.env.DATABASEURI || 'mongodb://127.0.0.1:27017/storie';
+const URI               = getConnectionURI();
 const client            = new MongoClient(URI, {
   useUnifiedTopology: true
 });
 
+function getConnectionURI() {
+  console.log(process.env.NODE_ENV);
+  if (process.env.NODE_ENV === "development") {
+    return 'mongodb://127.0.0.1:27017/storie';
```
Comment (christopher-dodge) [inline server/database/connection.js]: This needs to point to "dev_storie" database not "storie"

File: server/database/connection.js [hunk 2]
```diff
@@ -1,35 +1,44 @@
 const express           = require('express');
 const _dotenv           = require('dotenv').config();
```
Comment (krab7191) [inline server/database/connection.js]: Can delete these 2 unused imports

File: client/package.json [hunk 3]
```diff
@@ -7,9 +7,11 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "axios": "^0.20.0",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "react-scripts": "3.4.0"
+    "react-scripts": "3.4.0",
+    "typescript": "^4.0.3"
```
Comment (krab7191) [inline client/package.json:14]: Looks like you added TS and a config but no ts-node, tsc, etc to your scripts. Make sure you want to use TypeScript before setting it up, or just remove this dependency.

File: server/database/connection.js [hunk 4]
```diff
@@ -1,35 +1,44 @@
...
 const {MongoClient}     = require('mongodb');
-const URI               = process.env.DATABASEURI || 'mongodb://127.0.0.1:27017/storie';
+const URI               = getConnectionURI();
 const client            = new MongoClient(URI, {
   useUnifiedTopology: true
 });
 
+function getConnectionURI() {
+  console.log(process.env.NODE_ENV);
```
Comment (krab7191) [inline server/database/connection.js:8]: This is a good candidate for destructuring out NODE_ENV to reduce code. Maybe:

```
const { NODE_ENV } = process.env;
console.log(NODE_ENV);
if (NODE_ENV === 'development')
etc . . . 
```

**Comment by krab7191** [comment]
Looks like `BlogForm.test.tsx` needs to be renamed to a `.jsx` extension to make Create React App happy

File: server/database/connection.js (hunk 2 above)
Comment (christopher-dodge) [reply server/database/connection.js]: Why is line 2 not being used for the getConnectionURI function validation on lines 9-13?

File: client/package.json (hunk 3 above)
Comment (christopher-dodge) [reply client/package.json:14]: Ah, yep. Had it in there to fix an error. Just changed the BlogForm.test.tsx to .jsx and removed typescript and config. All good now.

**Pull request merged**